Source/SvgElement.cs

```diff
@@ -129,12 +129,23 @@ protected internal string ElementName
             {
                 if (string.IsNullOrEmpty(this._elementName))
                 {
+#if USE_SOURCE_GENERATORS
+                    if (SvgElements.ElementNames.TryGetValue(this.GetType(), out var elementName))
+                    {
+                        this._elementName = elementName;
+                    }
+                    else if (this is SvgDocument)

```

Comment:
Shouldn't this special case be added in the Generator?.
So that this Code Path stays

 if (SvgElements.ElementNames.TryGetValue(this.GetType(), out var elementName))
{
    this._elementName = elementName;
}

Reply:
SvgDocument does not have SvgElement attribute set, instead the attitude is used on SvgFragment so there would be duplicate im dictionary. I think the way is SvgDocument used this is better place for that exception to attribute convention used.

Reply:
That makes sense. So I agree that this is the better solution

Reply:
Added comment.